_shared_content/develop/guides/automation/create_a_module.md

```diff
@@ -208,6 +206,49 @@ class Request(Action):  # (4)!
 7. The `error` method will mark the action as failed and send back the error to the API.
 8. Finally, if everything went well we can return the results. The base action will take care of sending it back to the playbook API.
 
+## Test your code 
+Make sure you test your code with unitary and integration tests. In the following code, 
+we will test `get request`our previous example's action code. In practice, you have to cover all the use cases of the module. 
+```python
+import pytest
+import requests_mock
+from pydantic import HttpUrl
+from your_module import Request, RequestArguments, Response  # Adjust the import according to your module structure
+
+@requests_mock.Mocker()
+def test_get_request(mock_request):
+    # Mock the HTTP response
+    mock_response = {
+        'status_code': 200,
+        'headers': {'Content-Type': 'application/json'},
+        'text': 'Success'
+    }
+    mock_request.get('http://example.com', status_code=mock_response['status_code'],
+                     headers=mock_response['headers'], text=mock_response['text'])
+
+    # Create the arguments
+    arguments = RequestArguments(
+        url='http://example.com',
+        headers=None,
+        method='get'
+    )
+
+    # Instantiate and run the action
+    action = Request()
+    result = action.run(arguments)
+
+    # Assert the response
+    assert result.status_code == mock_response['status_code']
+    assert result.headers == mock_response['headers']
+    assert result.text == mock_response['text']
+
+
+```
+To run your tests, Poetry will be used for dependency installation. You can install all the necessary dependencies and run your test with the following command:
+```shell
+poetry run test_request.py
+```
```

Comment:
```suggestion
```
To effectively manage dependencies and run your tests, you should use Poetry. For more detailed information, you can check the [Poetry documentation](https://python-poetry.org/docs/). Additionally, you can look at the tests in the existing module.